code/modules/clothing/spacesuits/plasmamen.dm

```diff
@@ -210,6 +219,20 @@
 	user.update_inv_head() //So the mob overlay updates
 	update_button_icons(user)
 
+/obj/item/clothing/head/helmet/space/plasmaman/proc/smash_headlamp()
+	if(!lamp_functional)
+		return
+	if(!helmet_on)
+		return
```

Comment:
This shouldn't be in this proc, it is already done before calling smash_headlamp() in the relevant cases.